deployment/cdk/opensearch-service-migration/lib/stack-composer.ts

```diff
@@ -150,64 +193,221 @@ export class StackComposer {
+            fetchMigrationStack = new FetchMigrationStack(scope, "fetchMigrationStack", {
                 vpc: networkStack.vpc,
                 dpPipelineTemplatePath: dpPipelineTemplatePath,
                 sourceEndpoint: sourceClusterEndpoint,
-                targetEndpoint: process.env.MIGRATION_DOMAIN_ENDPOINT!,
-                stackName: `OSServiceHistoricalCDKStack-${stage}-${region}`,
+                stackName: `OSMigrations-${stage}-${region}-FetchMigration`,
                 description: "This stack contains resources to assist migrating historical data to an OpenSearch Service domain",
+                stage: stage,
+                defaultDeployId: defaultDeployId,
+                ...props,
+            })
+            fetchMigrationStack.addDependency(openSearchStack)
+            fetchMigrationStack.addDependency(migrationStack)
+            this.stacks.push(fetchMigrationStack)
+        }
+
+        let captureProxyESStack
+        if (captureProxyESServiceEnabled && networkStack && mskUtilityStack) {
+            captureProxyESStack = new CaptureProxyESStack(scope, "capture-proxy-es", {
+                vpc: networkStack.vpc,
+                stackName: `OSMigrations-${stage}-${region}-CaptureProxyES`,
+                description: "This stack contains resources for the Capture Proxy/Elasticsearch ECS service",
+                stage: stage,
+                defaultDeployId: defaultDeployId,
                 ...props,
             })
-            historicalCaptureStack.addDependency(opensearchStack)
-            this.stacks.push(historicalCaptureStack)
+            captureProxyESStack.addDependency(mskUtilityStack)
+            this.stacks.push(captureProxyESStack)
         }
 
+        let trafficReplayerStack
+        if ((trafficReplayerServiceEnabled && networkStack && openSearchStack && migrationStack && mskUtilityStack) || (addOnMigrationDeployId && networkStack)) {
+            trafficReplayerStack = new TrafficReplayerStack(scope, `traffic-replayer-${deployId}`, {
+                vpc: networkStack.vpc,
+                enableClusterFGACAuth: trafficReplayerEnableClusterFGACAuth,
+                addOnMigrationDeployId: addOnMigrationDeployId,
+                customTargetEndpoint: trafficReplayerTargetEndpoint,
+                customKafkaGroupId: trafficReplayerGroupId,
+                extraArgs: trafficReplayerExtraArgs,
+                enableComparatorLink: trafficComparatorServiceEnabled,
+                stackName: `OSMigrations-${stage}-${region}-${deployId}-TrafficReplayer`,
+                description: "This stack contains resources for the Traffic Replayer ECS service",
+                stage: stage,
+                defaultDeployId: defaultDeployId,
+                ...props,
+            })
+            if (mskUtilityStack) trafficReplayerStack.addDependency(mskUtilityStack)
+            if (migrationStack) trafficReplayerStack.addDependency(migrationStack)
```

Comment:
Please never put an if statement in w/out brackets! (see Apple's gotofail bug for one reason)
I'm even OK w/ putting them on one line when it helps with readability, but I'd still want to see brackets

Reply:
Don't know if I completely agree with the sentiment, but will agree enough that it is less error prone for future iterations